tdns/agent_utils.go

```
@@ -0,0 +1,612 @@
+					rrset, err := RecursiveDNSQueryWithServers(dns.Fqdn(targetName), dns.TypeSVCB, timeout, retries, resolvers)
+					if err != nil {
+						log.Printf("LocateAgent: error response to SVCB query: %v", err)
+						continue // Try the next transport
+					}
+
+					// Process SVCB response
+					if rrset == nil {
+						log.Printf("LocateAgent: response to %s SVCB contained zero RRs", targetName)
+						continue
+					}
+					for _, rr := range rrset.RRs {
```

golangci-lint (1.62.2)

SA5011: possible nil pointer dereference

(staticcheck)